requirements/specifications/lifecycle/index.md

```diff
@@ -287,189 +378,301 @@ manifest's releaseStatus is not developer.
-this section does not need to happen.
+> The platform **MUST** dispatch the `Lifecycle.onRequestDeactivate`
+> notification to the app, and wait for `appDeactivateTimeout` milliseconds
+> for either a `Lifecycle.deactivateResult` or `Lifecycle.deactivateError`
+> call in response.
+>
+> Once the platform receives the `deactivateResult` call, then the app
+> may be moved to the `RUNNING` state.
+>
+> If the app times out or makes a `deactivateError` call, then the app
+> **MUST** be terminated.
 
-At this point, the app **MUST** be in either the foreground or inactive
-state.
+During the `deactivate()` transition, the app **MUST** deallocate any
+A/V decoders.
 
-Finally, the platform **MUST** transition the app to the background
-state (see '[Background](#24-background)' above).
+The platform **MAY** begin to transition your app out of view as soon as
+`deactivate()` is called.
 
-### 3.5. Suspending an app
+### 4.5. Suspending an app
 
-Suspending an app transitions it to the suspended state, so that it is
-no longer using resources on the device.
+Suspending an app transitions it to the `SUSPENDED` state, where it is
+no longer allowed to use graphics composition, and is expected to consume
+less CPU and RAM.
 
-The platform may suspend apps for any number of reasons that are out of
-scope for this document. However, it is the platform's decision to
-suspend an app, not the app itself.
+The platform may suspend apps in order to free up memory, or for any
+number of reasons that are out of scope for this document. However, it
+is the platform's decision to suspend an app, not the app itself.
 
 To suspend an app, platforms **MUST** use the following process.
 
-If an app is already in the suspended state, then it is already
+If an app is in the `ACTIVE` state then it cannot yet be suspended, and
+**MUST** be deactivated first.
+
+If an app is in the `SLEEPING` state then it cannot be suspended and there
+is no need to do anything else. The platform **MUST NOT** dispatch any
+*additional* lifecycle notifications when attempting to suspend an app
+that is already in the `SLEEPING` state and the remainder of this section
+does not apply.
+
+If an app is already in the `SUSPENDED` state, then it is already
 suspended and there is no need to do anything else. The platform **MUST
 NOT** dispatch any *additional* lifecycle notifications when attempting
 to suspend an app that is already in the suspended state and the
-remainder of this section does not need to happen.
+remainder of this section does not apply.
 
-If the app is not loaded, or is not in the inactive state, then it
+If the app is not loaded, or is not in the `RUNNING` state, then it
 cannot be suspended, and the remainder of this section does not need to
 happen.
 
-At this point, the app **MUST** be in the inactive state.
+At this point, the app **MUST** be in the `RUNNING` state.
+
+If an app provides the `xrn:firebolt:capability:lifecycle:application`
```

Comment:
Do we need this as an "If"? The app MUST provide this right?

Reply:
Remove all of these "Ifs"